src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,159 @@
+            hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithExpression(subExpression)
+                .NormalizeWhitespace().WithLeadingTrivia(indentation); // replace Host/WebHost with identifierMethod
+
+            if (originalInvocation.ArgumentList.Arguments.Count == 0)
+            {
+                return editor.OriginalDocument;
+            }
+
+            var initArgumentExpr = originalInvocation.ArgumentList.Arguments[0].Expression; // builder => { }
+
+            if (initArgumentExpr is not LambdaExpressionSyntax lambdaExpr)
+            {
+                return editor.OriginalDocument;
+            }
+
+            if (lambdaExpr.Block != null)
+            {
+                var lambdaStatements = lambdaExpr.Block.Statements;
+                foreach (var statement in lambdaStatements)
+                {
+                    if (statement is not ExpressionStatementSyntax currentStatement
+                        || currentStatement.Expression is not InvocationExpressionSyntax expr)
+                    {
+                        return editor.OriginalDocument;
+                    }
+
+                    var argument = expr.ArgumentList; // arguments of builder.{method_name}({arguments})
+
+                    if (expr.Expression is not MemberAccessExpressionSyntax bodyExpression) //builder.{method_name} 
+                    {
+                        return editor.OriginalDocument;
+                    }
+
+                    var method = bodyExpression.Name; // method_name
+
+                    hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithName(method);
+                    originalInvocation = originalInvocation.Update(hostBasedInvocationMethodExpr, argument);
+                    hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithExpression(originalInvocation);
+
+                }
+            }
+
+            else if (lambdaExpr.ExpressionBody != null)
+            {
+                if (lambdaExpr.ExpressionBody is not InvocationExpressionSyntax body)
+                {
+                    return editor.OriginalDocument;
+                }
+
+                var argument = body.ArgumentList;
+
+                if (body.Expression is not MemberAccessExpressionSyntax bodyExpression)
+                {
+                    return editor.OriginalDocument;
+                }
+
+                var method = bodyExpression.Name;
+
+                hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithName(method);
+                originalInvocation = originalInvocation.WithExpression(hostBasedInvocationMethodExpr).WithArgumentList(argument);
+            }
+
+            else
+            {
+                return editor.OriginalDocument;
+            }
+
+            editor.ReplaceNode(diagnosticTarget, originalInvocation);
+        }
+
+        return editor.GetChangedDocument();
```

Comment:
I learned from https://github.com/dotnet/aspnetcore/pull/42757 that we should avoid using the document editor. Check out [this file](https://github.com/dotnet/aspnetcore/blob/f5932c1af29f0de3951a85b649b72279f575affc/src/Framework/AspNetCoreAnalyzers/src/CodeFixes/DetectMismatchedParameterOptionalityFixer.cs) for more info.